corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -957,44 +953,44 @@ def _get_detail_from_module(self, module, detail_type):
     Update references in the nodeset of the given datum, if necessary
 
-    e.g. "instance('casedb')/casedb/case[@case_type='guppy']
-                                        [@status='open']
-                                        [index/parent=instance('commcaresession')/session/data/parent_id]"
+    e.g. instance('casedb')/casedb/case
+             [@case_type='guppy']
+             [@status='open']
+             [index/parent=instance('commcaresession')/session/data/parent_id]
+
     is updated to
-            "instance('casedb')/casedb/case[@case_type='guppy']
-                                        [@status='open']
-                                        [index/parent=instance('commcaresession')/session/data/case_id]"
-                                                                                                ^^^^^^^
-    because the case referred to by "parent_id" in the child module has the ID "case_id" in the parent
-    module.
+         instance('casedb')/casedb/case[@case_type='guppy']
+             [@status='open']
+             [index/parent=instance('commcaresession')/session/data/case_id]
+                                                                    ^^^^^^^
+    because the case referred to by "parent_id" in the child module has the ID
+    "case_id" in the parent module.
     """
-    def _apply_change_to_datum_attr(datum, attr, change):
-        xpath = getattr(datum, attr, None)
-        if xpath:
-            old = session_var(change['old_id'])
-            new = session_var(change['new_id'])
-            setattr(datum, attr, xpath.replace(old, new))
-
-    datum = datum_meta.datum
-    action = datum_meta.action
-    if action:
-        if hasattr(action, 'case_indices'):
-            # This is an advanced module
-            for case_index in action.case_indices:
-                if case_index.tag in changed_ids:
-                    # update any reference to previously changed datums
-                    for change in changed_ids[case_index.tag]:
-                        _apply_change_to_datum_attr(datum, 'nodeset', change)
-                        _apply_change_to_datum_attr(datum, 'function', change)
-        else:
-            if 'basic' in changed_ids:
-                for change in changed_ids['basic']:
-                    _apply_change_to_datum_attr(datum, 'nodeset', change)
-                    _apply_change_to_datum_attr(datum, 'function', change)
+    old = session_var(old_id)
+    new = session_var(new_id)
+
+    for datum in datums:
+        for attr in ['nodeset', 'function']:
+            xpath = getattr(datum, attr, None)
+            if xpath:
+                setattr(datum, attr, xpath.replace(old, new))
+
+
+def _get_datums_by_case_tag(datums):
+    ret = defaultdict(list)
+    for datum in datums:
+        if datum.action:
+            if hasattr(datum.action, 'case_indices'):
+                # This is an advanced module
+                for case_index in datum.action.case_indices:
```

Comment:
That's tricky actually, since we need the `else` part of this block if `action` isn't set.  It could be
```python
for case_index in getattr(datum.action, 'case_indices', [CaseIndex(tag='basic')]):
```
But that also seems kinda messy